arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,65 @@
 #include <Smartcar.h>
 
+const int fSpeed   = 30;  // 30% of the full speed forward
+const int bSpeed   = -30; // 30% of the full speed backward
+const int lDegrees = -75; // degrees to turn left
+const int rDegrees = 75;  // degrees to turn right
+
+const int TRIGGER_PIN           = 6; // D6
+const int ECHO_PIN              = 7; // D7
+const unsigned int MAX_DISTANCE = 100;
+
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+
+SimpleCar car(control);
+
+
+void handleInput() { // handle serial input if there is any
+    if (Serial.available()) {
+        char input = Serial.read(); // read everything that has been received so far and log down
+                                    // the last entry
+        switch (input) {
+        case 'a': // rotate counter-clockwise going forward
+            car.setSpeed(fSpeed);
+            car.setAngle(lDegrees);
+            break;
+        case 's': // turn clock-wise
+            car.setSpeed(fSpeed);
+            car.setAngle(rDegrees);
+            break;
+        case 'w': // go ahead
+            car.setSpeed(fSpeed);
+            car.setAngle(0);
+            break;
+        case 'd': // go back
+            car.setSpeed(bSpeed);
+            car.setAngle(0);
+            break;
+        default: // if you receive something that you don't know, just stop
+            car.setSpeed(0);
+            car.setAngle(0);
+        }
+    }
+}
+
 void setup() {
-  // put your setup code here, to run once:
+    Serial.begin(9600);
 }
 
 void loop() {
-  // put your main code here, to run repeatedly:
-}
+    handleInput();
+
+    if (front.getDistance() < 50 && front.getDistance() != 0) 
```

Comment:
What happens if the first `getDistance` returns `0` and the second one `100`? The car will stop despite not having detected a close-enough obstacle.

Reply:
Yeah I didn't consider that
Would it be a good idea to instead store a variable holding 1 getDistance value and compare that instead?

Reply:
;)